### PR TITLE
Sync template files with upstream.

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

File: chef/cookbooks/swift/templates/default/container-server.conf.erb [hunk 1]
```diff
@@ -11,7 +11,6 @@ user = <%= node[:swift][:user] %>
 #
 # Use an integer to override the number of pre-forked processes that will
 # accept connections.
-# workers = auto
 workers = 2
```
Comment (vuntz) [inline chef/cookbooks/swift/templates/default/container-server.conf.erb]: In some other barclamps, we keep the default commented values. Not sure if we want to do this here?

File: chef/cookbooks/swift/templates/default/object-server.conf.erb [hunk 2]
```diff
@@ -13,7 +13,6 @@ user = <%= node[:swift][:user] %>
 #
 # Use an integer to override the number of pre-forked processes that will
 # accept connections.
-# workers = auto
 workers = 2
```
Comment (vuntz) [inline chef/cookbooks/swift/templates/default/object-server.conf.erb]: (same comment here)

File: chef/cookbooks/swift/templates/default/object-server.conf.erb [hunk 3]
```diff
@@ -125,6 +127,13 @@ log_requests = true
 # an abort to occur.
 # replication_failure_threshold = 100
 # replication_failure_ratio = 1.0
+#
+# Use splice() for zero-copy object GETs. This requires Linux kernel
+# version 3.0 or greater. If you set "splice = yes" but the kernel
+# does not support it, error messages will appear in the object server
+# logs at startup, but your object servers should continue to function.
+#
+# splice = no
```
Comment (vuntz) [inline chef/cookbooks/swift/templates/default/object-server.conf.erb]: Should we switch that to yes? (in a separate commit)

File: chef/cookbooks/swift/templates/default/proxy-server.conf.erb [hunk 4]
```diff
@@ -316,10 +320,20 @@ signing_dir = /var/cache/swift/keystone-signing
...
+# For backwards compatibility, keystoneauth will match names in cross-tenant
+# access control lists (ACLs) when both the requesting user and the tenant
+# are in the default domain i.e the domain to which existing tenants are
+# migrated. The default_domain_id value configured here should be the same as
+# the value used during migration of tenants to keystone domains.
+# default_domain_id = default
+# For a new installation, or an installation in which keystone projects may
+# move between domains, you should disable backwards compatible name matching
+# in ACLs by setting allow_names_in_acls to false:
+# allow_names_in_acls = true
```
Comment (vuntz) [inline chef/cookbooks/swift/templates/default/proxy-server.conf.erb]: I'm unsure if we should care about changing this one or not?

Comment (aplanas) [inline chef/cookbooks/swift/templates/default/proxy-server.conf.erb]: I don't foresee the implications of multidomains keystone, but looks like that the recomended value for new installations  is indeed `false`.  If we do that, we need to take care that for an update (cloud 4 -> 5) we need to use the current value of `true`.

File: chef/cookbooks/swift/templates/default/container-server.conf.erb (hunk 1 above)
Comment (aplanas) [inline chef/cookbooks/swift/templates/default/container-server.conf.erb]: Makes sense to do this also in the parametrized values?

I think that is this only one of the fews conf files in swift that reflect the default value, but makes sense for me.

**Comment by aplanas** [comment]
* Document default faulse in configuration files
* Enable splice option in object-server.conf.erb
* Set allow_names_in_acls to false, this is not backward compatible !!

**Comment by vuntz** [comment]
+1

**Comment by jsuchome** [comment]
+1

**Comment by mjura** [comment]
Looks good to me :+1:

**Pull request merged**